client/packages/common/src/ui/layout/tables/material-react-table/useBaseMaterialTable.tsx

```diff
@@ -9,19 +9,29 @@ import {
   CheckboxEmptyIcon,
   CheckboxIndeterminateIcon,
 } from '@common/icons';
+import {
+  getSavedTableState,
+  // resetSavedTableState,
+  useTableLocalStorage,
+} from './useTableLocalStorage';
+import { useRef } from 'react';
 
 export interface BaseTableConfig<T extends MRT_RowData>
   extends MRT_TableOptions<T> {
+  tableId: string; // key for local storage
   onRowClick?: (row: T) => void;
   isLoading: boolean;
 }
 
 export const useBaseMaterialTable = <T extends MRT_RowData>({
+  tableId,
   isLoading,
   onRowClick,
   state,
   ...tableOptions
 }: BaseTableConfig<T>) => {
+  const initialState = useRef(getSavedTableState(tableId));
```

Comment:
Made it a ref as it never changes after initial load. Could also memoise, but I think this is slightly more efficient and robust.